gui/src/JobSelector.tsx

```diff
@@ -0,0 +1,706 @@
+            },
+            {
+                id: "submitEntity",
+                primaryLabel: "Submitted entity",
+                secondaryLabel: "ID",
+                columnIndex: 3,
+                showPrimaryLabel: displayEntityLabel,
+                showSecondayLabel: displayEntityId,
+            },
+            {
+                id: "createdDt",
+                primaryLabel: "Creation date",
+                columnIndex: 5,
+                showPrimaryLabel: displayDate,
+                showSecondayLabel: displayDate,
+            },
+            {
+                id: "status",
+                primaryLabel: "Status",
+                columnIndex: 6,
+                showPrimaryLabel: true,
+                showSecondayLabel: true,
+            },
+        ],
+        [displayDate, displayEntityId, displayEntityLabel, displayJobId, displaySubmitId]
+    );
+
+    const handleClick = useCallback((event: React.MouseEvent<HTMLElement>) => {
+        const { id = "" } = event.currentTarget?.dataset || {};
+        setSelected((oldSelected) => {
+            const selectedIndex = oldSelected.indexOf(id);
+            let newSelected: string[] = [];
+
+            if (selectedIndex === -1) {
+                newSelected = newSelected.concat(oldSelected, id);
+            } else if (selectedIndex === 0) {
+                newSelected = newSelected.concat(oldSelected.slice(1));
+            } else if (selectedIndex === oldSelected.length - 1) {
+                newSelected = newSelected.concat(oldSelected.slice(0, -1));
+            } else if (selectedIndex > 0) {
+                newSelected = newSelected.concat(
+                    oldSelected.slice(0, selectedIndex),
+                    oldSelected.slice(selectedIndex + 1)
+                );
+            }
+            return [...newSelected];
+        });
+    }, []);
+
+    const handleSelectAllClick = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (event.target.checked) {
+                const newSelected = jobRows?.map((n) => n[0]);
+                setSelected(newSelected || []);
+                return;
+            }
+            setSelected([]);
+        },
+        [jobRows]
+    );
+
+    const handleCancelJobs = useCallback(
+        (event: React.MouseEvent<HTMLElement>) => {
+            const { id = "" } = event.currentTarget?.dataset || {};
+            dispatch(createSendActionNameAction(props.id, module, props.onJobAction, id));
+        },
+        [dispatch, module, props.id, props.onJobAction]
+    );
+    const handleDeleteJobs = useCallback(
+        (event: React.MouseEvent<HTMLElement>) => {
+            const { id = "" } = event.currentTarget?.dataset || {};
+            dispatch(createSendActionNameAction(props.id, module, props.onJobAction, id));
+        },
+        [dispatch, module, props.id, props.onJobAction]
+    );
+
+    const selectedAllowedCancelJobs = useMemo(
```

Comment:
this should be a boolean
and false if one of the selected row is not cancellable
use jobRows.some()